.github/workflows/setup_git.sh

```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env sh
+git config --global user.email "spack@example.com"
+git config --global user.name "Test User"
+# With fetch-depth: 0 we have a remote develop
+# but not a local branch. Don't do this on develop
+if [ "$(git branch --show-current)" != "develop" ]
+then
+  git branch develop origin/develop
+fi
```

Comment:
To reviewers: please double check that this will prevent us from doing `git branch develop origin/develop` while currently on `develop`.

Reply:
`--show-current` is supported for 2.22 and up. So this looks fine (including the `sh` syntax).

When do you expect this to be false? I assume that if we were to `git branch develop origin/develop` when there was already a develop branch, the command would fail. Would it also be possible to do this with `-f` like `git branch -f develop origin/develop`?

Reply:
> When do you expect this to be false?

This will be false as soon as we merge a PR and retrigger CI on a push to `develop`. 

> I assume that if we were to git branch develop origin/develop when there was already a develop branch, the command would fail

Indeed the command fails with non-zero exit code, and that would cause the entire workflow to fail.